src/cdk/tree/control/base-tree-control.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {SelectionModel} from '@angular/cdk/collections';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {Observable} from 'rxjs/Observable';
+import {TreeControl} from './tree-control';
+
+/** Base tree control. It has basic toggle/expand/collapse operations on a single node. */
+export abstract class BaseTreeControl<T> implements TreeControl<T> {
+  /** Saved node for `expandAll` action. */
+  nodes: T[];
+
+  /** Expansion info: the changes */
+  expansionChange = new BehaviorSubject<T[]>([]);
+
+  /** A selection model with multi-selection to track expansion status. */
+  expansionModel: SelectionModel<T> = new SelectionModel<T>(true);
+
+  /** Toggles one single node. Expands a collapsed node or collapse an expanded node. */
+  toggle(node: T) {
+    this.expansionModel.toggle(node);
+    this.expansionChange.next(this.expansionModel.selected);
+  }
+
+  /** Expands one single node. */
+  expand(node: T) {
+    this.expansionModel.select(node);
+    this.expansionChange.next(this.expansionModel.selected);
```

Comment:
You should be able to replace the `expansionChange` event with the `onChange` from the `SelectionModel`.